nucleus/map_label/MapLabelManager.cpp

```diff
@@ -0,0 +1,223 @@
+    // set everything to 0
+    memset(m_font_bitmap, 0, m_font_atlas_size.width() * m_font_atlas_size.height() * channels);
+
+    const int outline_bit_count = 4 * (m_font_outline.x - 1) * (m_font_outline.y - 1) + 2 * (m_font_outline.x - 1) + 2 * (m_font_outline.y - 1) + 1;
+    int* outline_mask = new int[outline_bit_count];
+    int index = 0;
+    for (int i = -m_font_outline.y + 1; i < m_font_outline.y; i++) {
+        for (int j = -m_font_outline.x + 1; j < m_font_outline.x; j++) {
+            outline_mask[index++] = (i * m_font_atlas_size.width() + j) * channels + 1;
+        }
+    }
+
+    // value that is added to the outline
+    constexpr uint8_t outline_damp_factor = 50;
+
+    for (int i = 0; i < lasty; ++i) {
+        for (int j = 0; j < m_font_atlas_size.width(); ++j) {
+            const uint8_t value = temp_bitmap[(i * m_font_atlas_size.width() + j)];
+            const int current_index = (i * m_font_atlas_size.width() + j) * channels;
+            m_font_bitmap[current_index] = value;
+
+            // add the outline to the outline channel
+            if (value != 0) {
+                // if the current pixel is part of the font itself -> be fully visible
+                m_font_bitmap[current_index + 1] = 255;
+                // calculate the outline for the outer parts
+                for (int k = 0; k < outline_bit_count; k++) {
+                    // increase the value by a factor
+                    uint8_t current_outline_value = m_font_bitmap[outline_mask[k] + current_index] + outline_damp_factor;
+                    if (current_outline_value < outline_damp_factor) // clamp the value (since we are working with uint8 -> values over 255 will flow over -> we therefore check for a value below the damp_factor)
+                        current_outline_value = 255;
+
+                    m_font_bitmap[outline_mask[k] + current_index] = current_outline_value;
+                }
+            }
+        }
+    }
+
+    delete[] outline_mask;
+}
+
+void MapLabelManager::create_font()
+{
+    // load ttf file
+    QFile file(":/fonts/SourceSans3-Medium.ttf");
+    const auto open = file.open(QIODeviceBase::OpenModeFlag::ReadOnly);
+    assert(open);
+    const QByteArray data = file.readAll();
+
+    // init font and get info about the dimensions
+    const auto font_init = stbtt_InitFont(&m_fontinfo, reinterpret_cast<const uint8_t*>(data.constData()), stbtt_GetFontOffsetForIndex(reinterpret_cast<const uint8_t*>(data.constData()), 0));
+    assert(font_init);
+
+    uint8_t* temp_bitmap = new uint8_t[m_font_atlas_size.width() * m_font_atlas_size.height()];
+
+    const std::vector<int> safe_chars = CharUtils::string_to_unicode_int_list(all_char_list);
+
+    float scale = stbtt_ScaleForPixelHeight(&m_fontinfo, MapLabel::font_size);
+    STBTT_memset(temp_bitmap, 0, m_font_atlas_size.width() * m_font_atlas_size.height()); // background of 0 around pixel
```

Comment:
STBTT_memset is just a preprocessor define.
therefore we can replace both, the new and this memset with
std::vector<uint8_t> temp_bitmap(m_font_atlas_size.width() * m_font_atlas_size.height(), uint8_t(0));

or, even better with a more easy to read nucleus::Raster implementation.